src/UI/Buyer/src/app/services/reflektion/reflektion.service.ts

```diff
@@ -0,0 +1,107 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { HSMeProduct } from "@ordercloud/headstart-sdk";
+import { ListPageWithFacets, MetaWithFacets } from "ordercloud-javascript-sdk";
+import { ProductSortOption } from "src/app/components/products/sort-products/sort-products.component";
+import { AppConfig } from "src/app/models/environment.types";
+import { ProductFilters } from "src/app/models/filter-config.types";
+
+@Injectable({
+    providedIn: 'root',
+  })
+  export class ReflektionService {
+    constructor(private http: HttpClient, private appConfig: AppConfig) {}
+        reflektionSortOptions: ProductSortOption[] = [
+            { label: "Name: A-Z", value: "name-asc" },
+            { label: "Name: Z-A", value: "name-desc" },
+            { label: "Price: High to Low", value: "price-desc" },
+            { label: "Price: Low to High", value: "price-asc" },
+            { label: "Rating: High to Low", value: "review-desc" },
+            { label: "Rating: Low to High", value: "review-asc" },
+            { label: "Reviews: High to Low", value: "review-desc" },
+            { label: "Reviews: Low to High", value: "review-asc" },
+            { label: "Featured", value: "featured-desc" },
+            
+        ];
+
+        async listReflektionProducts(userID: string, filters: ProductFilters) : Promise<ListPageWithFacets<HSMeProduct>> {
+            var body = this.buildRequest(userID, filters);
+            var resp = await this.http.post<any>(this.appConfig.reflektionUrl + "/api/search-rec/3", body, { headers: { Authorization: this.appConfig.reflektionAPIKey} }).toPromise();
+            var meProducts = { 
```

Comment:
Didn't you mention this API key is sensitive and should be stored in middleware? I would assume we'd want an endpoint that retrieves a reflektion token